### PR TITLE
Add One Collector Group ID

### PR DESCRIPTION
**Pull request by heya-msft**
Add didAddChannelUnit delegate

    * Change existing delegates param from MSChannelProtocol to MSChannelUnitProtocol in 
       MSChannelDelegate
    * Implement didAddChannelUnit in MSOneCollectorChannelDelegate
    * Add unit tests
    * Update unit test for MSChannelGroupDefaultTests

### CODE REVIEW

File: AppCenter/AppCenterTests/MSOneCollectorChannelDelegateTests.m [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+@property(nonatomic) MSOneCollectorChannelDelegate *sut;
+
+@end
+
+@implementation MSOneCollectorChannelDelegateTests
+
+- (void)setUp {
+  [super setUp];
+  self.sut = [[MSOneCollectorChannelDelegate alloc] init];
```
Comment (guperrot) [inline AppCenter/AppCenterTests/MSOneCollectorChannelDelegateTests.m]: new?

File: AppCenter/AppCenterTests/MSOneCollectorChannelDelegateTests.m [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+- (void)tearDown {
+  // Put teardown code here. This method is called after the invocation of each test method in the class.
+  [super tearDown];
```
Comment (guperrot) [inline AppCenter/AppCenterTests/MSOneCollectorChannelDelegateTests.m]: do we need that?

File: AppCenter/AppCenter/Internals/Channel/MSOneCollectorChannelDelegate.m [hunk 3]
```diff
@@ -1,10 +1,48 @@
-#import "MSOneCollectorChannelDelegate.h"
+#import "MSOneCollectorChannelDelegatePrivate.h"
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/Internals/Channel/MSOneCollectorChannelDelegate.m]: Please sort imports.

File: AppCenter/AppCenterTests/MSOneCollectorChannelDelegateTests.m [hunk 4]
```diff
@@ -0,0 +1,97 @@
+#import <XCTest/XCTest.h>
+#import "MSOneCollectorChannelDelegatePrivate.h"
```
Comment (jaelim-ms) [inline AppCenter/AppCenterTests/MSOneCollectorChannelDelegateTests.m]: Please sort imports.

File: AppCenter/AppCenterTests/MSOneCollectorChannelDelegateTests.m [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+@implementation MSOneCollectorChannelDelegateTests
+
+- (void)setUp {
+  [super setUp];
+  self.sut = [[MSOneCollectorChannelDelegate alloc] init];
+}
+
+- (void)tearDown {
+  // Put teardown code here. This method is called after the invocation of each test method in the class.
```
Comment (jaelim-ms) [inline AppCenter/AppCenterTests/MSOneCollectorChannelDelegateTests.m]: Empty space before comment.

**Pull request merged**